edu_edfi_airflow/dags/edfi_resource_dag.py

```diff
@@ -277,195 +356,458 @@ def build_change_version_task_group(self) -> TaskGroup:
         )
+        get_cv_operator >> failed_sentinel
 
+        return get_cv_operator
 
-    def build_dbt_var_increment_operator(self):
+    def build_change_version_update_operator(self,
+        task_id: str,
+        endpoints: List[str],
+        get_deletes: bool,
+        get_key_changes: bool,
+        **kwargs
+    ) -> PythonOperator:
         """
 
         :return:
         """
-        def short_circuit_update_variable(**kwargs):
-            """
-            Helper to build short-circuit logic into the update_variable callable if no new data was ingested.
-            :return:
-            """
-            from airflow.exceptions import AirflowSkipException
-
-            for task_id in kwargs['task'].upstream_task_ids:
-                if kwargs['ti'].xcom_pull(task_id):
-                    return update_variable(**kwargs)
-            else:
-                raise AirflowSkipException(
-                    "There is no new data to process using DBT. All upstream tasks skipped or failed."
-                )
-
         return PythonOperator(
-            task_id='increment_dbt_variable',
-            python_callable=short_circuit_update_variable,
+            task_id=task_id, 
+            python_callable=change_version.update_change_versions,
             op_kwargs={
-                'var': self.dbt_incrementer_var,
-                'value': lambda x: int(x) + 1,
+                'tenant_code': self.tenant_code,
+                'api_year': self.api_year,
+                'snowflake_conn_id': self.snowflake_conn_id,
+                'change_version_table': self.change_version_table,
+                
+                'edfi_change_version': airflow_util.xcom_pull_template(self.newest_edfi_cv_task_id),
+                'endpoints': endpoints,
+                'get_deletes': get_deletes,
+                'get_key_changes': get_key_changes,
             },
-            trigger_rule='all_done',
-            dag=self.dag
+            provide_context=True,
+            dag=self.dag,
+            **kwargs
         )
 
 
-    def build_edfi_to_snowflake_task_group(self,
-        resource : str,
-        namespace: str = 'ed-fi',
+    # Polymorphic Ed-Fi TaskGroups
+    @staticmethod
+    def xcom_pull_template_map_idx(task_ids, idx: int):
+        """
+        Many XComs in this DAG are lists of tuples. This overloads xcom_pull_template to retrieve a list of items at a given index.
+        """
+        return airflow_util.xcom_pull_template(
+            task_ids, suffix=f" | map(attribute={idx}) | list"
+        )
+    
+    @staticmethod
+    def xcom_pull_template_get_key(task_ids, key: str):
+        """
+        Many XComs in this DAG are lists of tuples. This converts the XCom to a dictionary and returns the value for a given key.
+        """
+        return airflow_util.xcom_pull_template(
+            task_ids, prefix="dict(", suffix=f").get('{key}')"
+        )
+
+    def build_default_edfi_to_snowflake_task_group(self,
```

Comment:
This is the default behavior. All endpoints are attempted, but those not found in the get_cv_operator XCom are skipped immediately. The final Snowflake copy is completed in a single bulk task.